cdc/sink/mq.go

```diff
@@ -308,7 +336,11 @@ func (k *mqSink) runWorker(ctx context.Context, partition int32) error {
 		}
 		if e.row == nil {
 			if e.resolvedTs != 0 {
-				if err := flushToProducer(codec.EncoderNeedAsyncWrite); err != nil {
+				op, err := encoder.AppendResolvedEvent(e.resolvedTs)
```

Comment:
Should we change the return value or `AppendResolvedEvent` to `codec.EncoderNeedAsyncWrite`, in some codec implement?

Reply:
Yes, we should. This has caused Kafka integration tests to fail.